imports/plugins/included/email-smtp/client/actions/settings.js

```diff
@@ -45,34 +57,42 @@ export default {
             "error"
           );
         }
-        return Alert({
-          title: i18next.t("app.success"),
-          text: i18next.t("mail.alerts.saveSuccess"),
-          type: "success",
-          timer: 1700
-        }).catch(() => null);
+        return Alerts.toast(i18next.t("app.success"), "success");
       });
     };
 
+    const shopId = Reaction.getPrimaryShopId();
+    const [opaqueShopId] = await getOpaqueIds([{ namespace: "Shop", id: shopId }]);
+
     // check if the settings work first
-    Meteor.call("email/verifySettings", settings, (error) => {
-      callback();
-      // if the connection fails
-      if (error) {
-        // eslint-disable-next-line promise/no-promise-in-callback
-        Alert({
-          title: i18next.t("mail.alerts.connectionFailed"),
-          text: i18next.t("mail.alerts.saveAnyway"),
-          type: "warning",
-          showCancelButton: true,
-          cancelButtonText: i18next.t("app.cancel"),
-          confirmButtonColor: "#DD6B55",
-          confirmButtonText: i18next.t("app.save")
-        }).then(() => save()).catch(() => true);
-      } else {
-        save();
+    await client.mutate({
+      mutation: verifySMTPEmailSettings,
+      variables: {
+        input: {
+          host: settings.host,
+          password: settings.password,
+          port: settings.port,
+          service: settings.service,
+          shopId: opaqueShopId,
+          user: settings.user
+        }
       }
-    });
+    })
+      .then(() => {
+        // eslint-disable-next-line promise/no-callback-in-promise
+        callback();
+        return save();
+      })
+      // eslint-disable-next-line promise/no-nesting
+      .catch(() => Alert({
+        title: i18next.t("mail.alerts.connectionFailed"),
+        text: i18next.t("mail.alerts.saveAnyway"),
+        type: "warning",
+        showCancelButton: true,
+        cancelButtonText: i18next.t("app.cancel"),
+        confirmButtonColor: "#DD6B55",
+        confirmButtonText: i18next.t("app.save")
+      }).then(() => save()).catch(() => true));
```

Comment:
If using async/await, it's best to stick with that and try/catch instead of also using `.then` and `.catch`.

```js
let shouldSave = true;
try {
    await client.mutate({
      mutation: verifySMTPEmailSettings,
      variables: {
        input: {
          host: settings.host,
          password: settings.password,
          port: settings.port,
          service: settings.service,
          shopId: opaqueShopId,
          user: settings.user
        }
      }
    });
} catch (error) {
  shouldSave = false;
}

if (!shouldSave) {
  try {
      await Alert({
        title: i18next.t("mail.alerts.connectionFailed"),
        text: i18next.t("mail.alerts.saveAnyway"),
        type: "warning",
        showCancelButton: true,
        cancelButtonText: i18next.t("app.cancel"),
        confirmButtonColor: "#DD6B55",
        confirmButtonText: i18next.t("app.save")
      })
  } catch (error) {
    callback();
    return;
  }
}

callback();
await save();
```

It's still a bit weird. If it doesn't work and what you have does work, then it's fine. We should eventually make a confirm Alert that returns true/false rather than throwing so we don't need so many try/catch.

Reply:
I was struggling with getting `try/catch` to work, and it just wasn't, which is why i'm doing the different `then...catch` in both places here. I'll try again with what you've posted here, but it was just being weird.